### PR TITLE
Alternate Debug Formatting

### PR DESCRIPTION
**Pull request by IsaacCloos**
Improve development experience by extending `alternate()` formatting options to the `grid` struct.

This includes:
1. new line formatting
2. width parameter
3. precision parameter

Tests included.

Using the `alternate()` flag in a predictable way helps achieve 👇

> Most of the functions std::Vec<T> offer are also implemented in grid and slightly modified for a 2D data object.

#### note:
In my testing I found that `str` and `String` don't respect the width/padding features of `format!()`. 

This results in most grids of strings (or compound types containing them) being doomed to look uneven.

If anyone knows how to address that I'd love to learn 😄

# criticism welcome

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -772,8 +772,23 @@ impl<T: fmt::Debug> fmt::Debug for Grid<T> {
...
         if self.cols > 0 {
-            for (i, _) in self.data.iter().enumerate().step_by(self.cols) {
-                write!(f, "{:?}", &self.data[i..(i + self.cols)]);
+            if f.alternate() {
+                write!(f, "\n");
+                for (i, _) in self.data.iter().enumerate().step_by(self.cols) {
+                    let mut row = self.data[i..(i + self.cols)].iter().peekable();
+                    write!(f, "    [");
+                    while let Some(item) = row.next() {
+                        write!(f, " {item:width$.precision$?}", width = f.width().unwrap_or(self.data.iter().map(|i| format!("{i:?}").len()).max().unwrap()), precision = f.precision().unwrap_or(2));
```
Comment (becheran) [inline src/lib.rs:781]: Hard to follow in one line. Line breaks would be great for improved readability

File: src/lib.rs [hunk 2]
```diff
@@ -1152,6 +1167,115 @@ mod test {
...
+        
+        let expected_output = 
+r#"[
+    [ Person { _name: "Vic", _precise_age: 24.50000 }, Person { _name: "Mr. Very Long Name", _precise_age: 1955.00000 }]
+    [ Person { _name: "Sam", _precise_age: 8.99950 }, Person { _name: "John Doe", _precise_age: 40.14000 }]
+]"#;
+
+        assert_eq!(expected_output, format!("{:#5.5?}", grid));
+    }
+
```
Comment (becheran) [inline src/lib.rs:1278]: Thx so much for adding tests. One edge case that just came into my mind is adding a pretty format test for empty grids

**Review by becheran — CHANGES_REQUESTED** [review]
Looks good. Thx. Could you please address the two minor suggestions? Otherwise a great improvement

File: src/lib.rs (hunk 1 above)
Comment (IsaacCloos) [reply src/lib.rs]: Formatted, removed from loop, commented! Thanks for the feedback, makes a huge difference :)

File: src/lib.rs (hunk 2 above)
Comment (IsaacCloos) [reply src/lib.rs:1306]: I added an empty grid test and also rearranged the other tests I added to match the format of yours.

I had `assert_eq(expected, actual)` whereas you had `assert_eq(actual, expected)` so I changed to match that format.

**Review by becheran — APPROVED** [review]
Looks good! 👍
